draft-ietf-quic-transport.md

```diff
@@ -1410,6 +1416,71 @@ a single packet.
+check is performed by attempting to set the ECN bits in the IP header to either
+ECT(0) or ECT(1) for packets generated by the endpoint, clients starting with
+the Initial packet ({{packet-initial}}) and servers with the Handshake packet
+(server) ({{packet-handshake}}). Note that Retry Packets ({{packet-retry}}) can
+be marked as ECT, but as they are not acknowledged by the peer, they provide no
+capability verification.
+
+The choice if ECT(0) and ECT(1) should be used is as per the guidelines in
+{{!RFC8311}}. ECT(0) is the default, if no ECN experiments are run. Upon
+reception of a packet with ECT(0), ECT(1) or ECN-CE marking by the peer
+endpoint, an ACK_ECN frame is transmitted back. This ACK_ECN frame indicates how
+many packets that are marked ECT(0), ECT(1) or ECN-CE. Thus the endpoint can
+determine if the endpoints and path is ECN capable.
+
+The ACK_ECN frame will, when received, confirm that the path direction supports
+ECN if the counters show a correct amount of packets received for a valid and
+expected counter combination. ECT marked packets can become remarked as CE along
+the path between the peers. The ACK_ECN counters are added together and compared
+to total number of ACKed packets, if the sum of the counters are equal too or
+larger than the number of ACKed packets then the path is ECN capable. This
+capability check thus has verified that one direction of the path between the
+peers is free from issues with ECN bleaching and that the application does not
+experience problems with access to the ECN field in the IP header. In this case
+the marking of the generated packets with ECT continues.
+
+If an ACK frame (not the ACK_ECN frame) is used to acknowledge reception of
+packets that was marked as ECT in the sender, or if the comparision of the
+counters with the total of acknolwedged packets indicates that not all packets
+arrive as ECT(0), ECT(1) or ECN-CE, then the path is determined as not ECN
+Capable.  If the endpoint determine the path as currently not ECN capable it
+SHALL stop marking the packets as ECT, and instead mark them as Not-ECT. If the
+connection is migrated then the ECN capability check is rerun as specified in
+{{ecn-connection-migration}}.
+
+It is expected that QUIC discards duplicate packets early, however if that is
+not the case **\[ED note, have not seen any clear statement in the drafts]**,
+then it should be verified that the number of ECT marked packets are equal to or
+larger that the amount of ECT marked packets that have been transmitted.
+
+### Continous Verification of ECN {#ecn-continous-verification}
+
+If the ECN capabiity check was successful and the endpoint continus to send ECT
+marked packets then continous verification is applied. This is to detect any
+cases when ECN field is bleached, that is, zeroed out by a network node, likely
+as the result of a routing changes since the ECN capability check.
+
+For each ACK_ECN frame that is received, the total number of ACKed packets are
```

Comment:
Suggestion: "For each received ACK_ECN frame, the total number of newly acknowledged packets can be compared to the total increase in ECN counters.  If the increase in ECN counters is larger, then an ECN failure has occurred and ECN should be disabled."

Reply:
Yes, except that larger needs to be less in this case.